### PR TITLE
[opentelemetry-instrumentation-openai] Capture cached prompt tokens on chat completions

### LINKED ISSUE #546: [openai] Capture cached prompt tokens on chat completions
> [!NOTE]
> This issue was generated with AI assistance and requires investigation and confirmation prior to being worked on.

Part of #541.

Extract and report cached prompt tokens for `chat.completions.create` and streaming chat responses.

- Extract from `usage.prompt_tokens_details.cached_tokens`.
- Semconv attribute: `gen_ai.usage.cache_read.input_tokens`.
- Affected files: `patch.py` and `chat_wrappers.py`.

### PR DESCRIPTION
**Pull request by rads-1996**
# Description

Fixes https://github.com/open-telemetry/opentelemetry-python-genai/issues/546

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce. List any relevant details for your test
configuration.

- [ ] Test A

# Checklist

See [CONTRIBUTING.md](https://github.com/open-telemetry/opentelemetry-python-genai/blob/main/CONTRIBUTING.md)
for the style guide, changelog guidance, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelog updated if the change requires an entry
- [x] Unit tests added
- [ ] Documentation updated

### CODE REVIEW

**Review by eternalcuriouslearner — APPROVED** [review]
LGTM!! One small ask, can we please add this to conformance tests and few new vcr tests if possible?

File: instrumentation/opentelemetry-instrumentation-genai-openai/tests/test_chat_completions.py [hunk 1]
```diff
@@ -80,6 +81,7 @@ def test_chat_completion_with_content(
         response.usage.prompt_tokens,
         response.usage.completion_tokens,
     )
+    assert_cache_attributes(spans[0], response.usage)
```
Comment (lmolkova) [inline instrumentation/opentelemetry-instrumentation-genai-openai/tests/test_chat_completions.py:85]: The test cassettes have `cached_tokens: 0`, so this only exercises attribute absence. Add tests with non-zero `cached_tokens` (sync, async, streaming) to verify `gen_ai.usage.cache_read.input_tokens` is emitted.

File: instrumentation/opentelemetry-instrumentation-genai-openai/tests/test_utils.py [hunk 2]
```diff
@@ -561,7 +562,11 @@ def assert_cache_attributes(span, usage):
     details = _get_usage_details(usage)
     assert details is not None
 
-    cached_tokens = getattr(details, "cached_tokens", None)
+    cached_tokens = (
```
Comment (lmolkova) [inline instrumentation/opentelemetry-instrumentation-genai-openai/tests/test_utils.py]: nit: Use `get_property_value(details, "cached_tokens")` from `.utils` here as well.

**Comment by rads-1996** [comment]
> LGTM!! One small ask, can we please add this to conformance tests and few new vcr tests if possible?

@eternalcuriouslearner Thank you for the feedback. I have addressed your comments and added the vcr tests.

**Review by eternalcuriouslearner — APPROVED** [review]

**Review by lmolkova — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-genai-openai/tests/test_chat_completions.py (hunk 1 above)
Comment (rads-1996) [reply instrumentation/opentelemetry-instrumentation-genai-openai/tests/test_chat_completions.py:85]: We do capture the non-zero cached_tokens in the cassettes to verify the attribute is emitted -

image

File: instrumentation/opentelemetry-instrumentation-genai-openai/tests/test_utils.py (hunk 2 above)
Comment (rads-1996) [reply instrumentation/opentelemetry-instrumentation-genai-openai/tests/test_utils.py]: Addressed.

File: instrumentation/opentelemetry-instrumentation-genai-openai/tests/test_chat_completions.py (hunk 1 above)
Comment (lmolkova) [reply instrumentation/opentelemetry-instrumentation-genai-openai/tests/test_chat_completions.py:85]: thanks, sorry I missed it!

**Review by lmolkova — APPROVED** [review]
